backend/tests/unit/test_url_validation.py

```diff
@@ -0,0 +1,161 @@
+"""Unit tests for TradeSafetyService URL validation."""
+
+import unittest
+
+from aioia_core.settings import OpenAIAPISettings
+
+from trade_safety.service import TradeSafetyService
+from trade_safety.settings import TradeSafetyModelSettings
+
+
+class TestURLValidation(unittest.TestCase):
+    """Test _is_url method for URL detection."""
+
+    def setUp(self):
+        """Set up test fixtures before each test method."""
+        openai_api = OpenAIAPISettings(api_key="sk-test-dummy-key")
+        model_settings = TradeSafetyModelSettings()
+        self.service = TradeSafetyService(openai_api, model_settings)
```

Comment:
`TradeSafetyService` 초기화 시 `TwitterService`가 `TWITTER_BEARER_TOKEN`을 필요로 해서, unit test에서 dummy token 설정이 필요합니다.


```suggestion
"""Unit tests for TradeSafetyService URL validation."""

import os
import unittest

from aioia_core.settings import OpenAIAPISettings

from trade_safety.service import TradeSafetyService
from trade_safety.settings import TradeSafetyModelSettings


class TestURLValidation(unittest.TestCase):
    """Test _is_url method for URL detection."""

    def setUp(self):
        """Set up test fixtures before each test method."""
        # Set dummy tokens for unit tests (no real API calls)
        os.environ["TWITTER_BEARER_TOKEN"] = "test-dummy-twitter-token"
        openai_api = OpenAIAPISettings(api_key="sk-test-dummy-key")
        model_settings = TradeSafetyModelSettings()
        self.service = TradeSafetyService(openai_api, model_settings)
```